pallets/randomness/src/lib.rs

```diff
@@ -0,0 +1,299 @@
+		// /// Weight information for extrinsics in this pallet.
+		// type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NotYetImplemented,
+		RequestCounterOverflowed,
+		NotSufficientDeposit,
+		CannotRequestPastRandomness,
+		InvalidRequestCannotBeFulfilledBeforeExpiry,
+		RequestDNE,
+		RequestCannotYetBeFulfilled,
+		OnlyRequesterCanIncreaseFee,
+		NewFeeMustBeGreaterThanOldFee,
+		RequestHasNotExpired,
+		RequestedRandomnessNotCorrectlyUpdated,
+		RequestExecutionOOG,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Randomness requested and request put in storage
+		RandomnessRequested {
+			id: RequestId,
+			refund_address: T::AccountId,
+			contract_address: T::AccountId,
+			fee: BalanceOf<T>,
+			salt: T::Hash,
+			info: RequestType<T>,
+		},
+		RequestFulfilled {
+			id: RequestId,
+		},
+		RequestFeeIncreased {
+			id: RequestId,
+			new_fee: BalanceOf<T>,
+		},
+		RequestExpirationExecuted {
+			id: RequestId,
+		},
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn request)]
+	/// Randomness requests not yet fulfilled or purged
+	pub type Requests<T: Config> =
+		StorageMap<_, Blake2_128Concat, RequestId, RequestState<T>, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn request_count)]
+	/// Number of randomness requests made so far, used to generate the next request's uid
+	pub type RequestCount<T: Config> = StorageValue<_, RequestId, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn current_epoch_index)]
+	/// Most recent epoch index, when it changes => update the epoch randomness
+	pub type CurrentEpochIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn current_block_randomness)]
+	/// Relay chain current block randomness
+	/// Some(randomness) or None if not updated
+	/// TODO: replace with ParentBlockRandomness once
+	/// https://github.com/paritytech/substrate/pull/11113 is merged
+	pub type CurrentBlockRandomness<T: Config> = StorageValue<_, Option<T::Hash>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn last_current_block_randomness)]
+	/// Last relay chain current block randomness
+	/// Some(randomness) or None if not updated
+	/// TODO: replace with LastParentBlockRandomness once
+	/// https://github.com/paritytech/substrate/pull/11113 is merged
+	pub type LastCurrentBlockRandomness<T: Config> = StorageValue<_, Option<T::Hash>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn one_epoch_ago_randomness)]
+	/// Relay chain one epoch ago randomness
+	/// Some(randomness) or None if not updated
+	pub type OneEpochAgoRandomness<T: Config> = StorageValue<_, Option<T::Hash>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn last_one_epoch_ago_randomness)]
+	/// Last relay chain one epoch ago randomness
+	/// Some(randomness) or None if not updated
+	pub type LastOneEpochAgoRandomness<T: Config> = StorageValue<_, Option<T::Hash>, ValueQuery>;
```

Comment:
Are `LastOneEpochAgoRandomness` and `OneEpochAgoRandomness` gonna be written/read separately? I am just wondering why have two storage items here isntead of one

Reply:
I don't even know why we wanted `LastOneEpochAgoRandomness` honestly. I'm going to remove it.